core/src/test/scala/integration/kafka/api/GroupCoordinatorIntegrationTest.scala

```diff
@@ -12,53 +12,212 @@
+      new ClusterConfigProperty(key = "offsets.topic.compression.codec", value = "1"),
+      new ClusterConfigProperty(key = "controlled.shutdown.enable", value = "false"),
+    )
+  )
+  def testGroupCoordinatorPropagatesOffsetsTopicCompressionCodec(): Unit = {
+    val logManager = cluster.brokers().asScala.head._2.logManager
+    val consumer = TestUtils.createConsumer(cluster.bootstrapServers())
+
+    try {
+      consumer.commitSync(Map(
+        new TopicPartition(Topic.GROUP_METADATA_TOPIC_NAME, 0) -> new OffsetAndMetadata(10, "")
+      ).asJava)
+
+      def getGroupMetadataLogOpt: Option[UnifiedLog] =
+        logManager.getLog(new TopicPartition(Topic.GROUP_METADATA_TOPIC_NAME, 0))
+
+      TestUtils.waitUntilTrue(() => getGroupMetadataLogOpt.exists(_.logSegments.asScala.exists(_.log.batches.asScala.nonEmpty)),
+        "Commit message not appended in time")
+
+      val logSegments = getGroupMetadataLogOpt.get.logSegments.asScala
+      val incorrectCompressionCodecs = logSegments
+        .flatMap(_.log.batches.asScala.map(_.compressionType))
+        .filter(_ != CompressionType.GZIP)
+
+      assertEquals(Seq.empty, incorrectCompressionCodecs, "Incorrect compression codecs should be empty")
+    } finally {
+      consumer.close()
+    }
   }
 
-  @ParameterizedTest
-  @ValueSource(strings = Array("zk", "kraft"))
-  def testGroupCoordinatorPropagatesOffsetsTopicCompressionCodec(quorum: String): Unit = {
-    val consumer = TestUtils.createConsumer(bootstrapServers())
-    val offsetMap = Map(
-      new TopicPartition(Topic.GROUP_METADATA_TOPIC_NAME, 0) -> new OffsetAndMetadata(10, "")
-    ).asJava
-    consumer.commitSync(offsetMap)
-    val logManager = brokers.head.logManager
-    def getGroupMetadataLogOpt: Option[UnifiedLog] =
-      logManager.getLog(new TopicPartition(Topic.GROUP_METADATA_TOPIC_NAME, 0))
-
-    TestUtils.waitUntilTrue(() => getGroupMetadataLogOpt.exists(_.logSegments.asScala.exists(_.log.batches.asScala.nonEmpty)),
-                            "Commit message not appended in time")
-
-    val logSegments = getGroupMetadataLogOpt.get.logSegments.asScala
-    val incorrectCompressionCodecs = logSegments
-      .flatMap(_.log.batches.asScala.map(_.compressionType))
-      .filter(_ != offsetsTopicCompressionCodec)
-    assertEquals(Seq.empty, incorrectCompressionCodecs, "Incorrect compression codecs should be empty")
-
-    consumer.close()
+  @ClusterTest(
+    types = Array(Type.KRAFT),
+    serverProperties = Array(
+      new ClusterConfigProperty(key = "group.coordinator.new.enable", value = "true"),
+      new ClusterConfigProperty(key = "group.coordinator.rebalance.protocols", value = "classic,consumer"),
+      new ClusterConfigProperty(key = "offsets.topic.num.partitions", value = "1"),
+      new ClusterConfigProperty(key = "offsets.topic.replication.factor", value = "1")
+    )
+  )
+  def testCoordinatorFailoverWithConsumerGroupRecordsAfterCompactingPartition(): Unit = {
```

Comment:
if we can, i think it makes more sense to split the test if possible which would make it easier to debug in the future if any fails.

Reply:
ditto, these configs could be found in `GroupCoordinatorConfig`.